katdal/test/test_chunkstore.py

```diff
@@ -128,26 +128,37 @@ def array_name(self, name):
         return name
 
     def put_and_get_chunk(self, var_name, slices):
+        """Put a single chunk into store, get it back and compare."""
         array_name = self.array_name(var_name)
         chunk = getattr(self, var_name)[slices]
         self.store.put_chunk(array_name, slices, chunk)
         chunk_retrieved = self.store.get_chunk(array_name, slices, chunk.dtype)
         assert_array_equal(chunk_retrieved, chunk,
                            "Error storing {}[{}]".format(var_name, slices))
 
-    def put_and_get_dask_array(self, var_name):
+    def put_and_get_dask_array(self, var_name, offset_in_chunks=()):
+        """Put dask array into store (with offset), get it back and compare."""
         array_name = self.array_name(var_name)
         array = getattr(self, var_name)
         chunks = generate_chunks(array.shape, array.dtype, array.nbytes / 10.)
+        offset = ()
+        if offset_in_chunks:
+            offset = tuple(np.cumsum(np.r_[0, ch])[of]
+                           for ch, of in zip(chunks, offset_in_chunks))
+            chunks = tuple(ch[of:] for ch, of in zip(chunks, offset_in_chunks))
+            offset_slice = tuple(slice(start, None) for start in offset)
+            array = array[offset_slice]
         divisions_per_dim = [len(c) for c in chunks]
         dask_array = da.from_array(array, chunks)
-        push = self.store.put_dask_array(array_name, dask_array)
+        push = self.store.put_dask_array(array_name, dask_array, offset)
         results = push.compute()
         assert_array_equal(results, np.tile(None, divisions_per_dim))
-        pull = self.store.get_dask_array(array_name, chunks, array.dtype)
+        pull = self.store.get_dask_array(array_name, chunks, array.dtype,
+                                         offset)
```

Comment:
This test is a bit weak, because it only tests passing the same offset to put and then fetch the data. For example, if `_add_offset_to_slices` was replaced with an identity function, I think the test would still pass. It may be best to test e.g. writing an array in several pieces, then pulling it all back in one go.

Reply:
Done, thanks, it's much better now.